src/query_engine/query_simplification.cpp

```diff
@@ -176,34 +259,59 @@ std::unique_ptr<BoolExpression> OrEx::simplify(const Database& db, const Databas
       return ret;
    }
 }
+OrExpression::OrExpression() {}
```

Comment:
Why are there implementations for `OrExpression` in this file? (same for `NOfExpression` below).